# Platzi Data Challenge
## Titanic challenge 1

In [1]:
import numpy as np
import pandas as pd

Import datasets

In [2]:
train_df = pd.read_csv('Data/train.csv')
test_df = pd.read_csv('Data/test.csv')
gender_submission_df = pd.read_csv('Data/gender_submission.csv')


Exploring data

In [3]:
train_df.describe

<bound method NDFrame.describe of      PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                 

In [4]:
train_df.shape

(891, 12)

In [5]:
test_df.describe

<bound method NDFrame.describe of      PassengerId  Pclass                                          Name  \
0            892       3                              Kelly, Mr. James   
1            893       3              Wilkes, Mrs. James (Ellen Needs)   
2            894       2                     Myles, Mr. Thomas Francis   
3            895       3                              Wirz, Mr. Albert   
4            896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
..           ...     ...                                           ...   
413         1305       3                            Spector, Mr. Woolf   
414         1306       1                  Oliva y Ocana, Dona. Fermina   
415         1307       3                  Saether, Mr. Simon Sivertsen   
416         1308       3                           Ware, Mr. Frederick   
417         1309       3                      Peter, Master. Michael J   

        Sex   Age  SibSp  Parch              Ticket      Fare Cabin Embarked 

In [6]:
gender_submission_df.describe

<bound method NDFrame.describe of      PassengerId  Survived
0            892         0
1            893         1
2            894         0
3            895         0
4            896         1
..           ...       ...
413         1305         0
414         1306         1
415         1307         0
416         1308         0
417         1309         0

[418 rows x 2 columns]>

Merging dataframes

In [7]:
secundary_df = pd.merge(test_df,gender_submission_df, how = 'inner', on = ['PassengerId'])
titanic_df = pd.concat([train_df,secundary_df])
titanic_df.describe

<bound method NDFrame.describe of      PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
413         1305         0       3   
414         1306         1       1   
415         1307         0       3   
416         1308         0       3   
417         1309         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                 

In [8]:
titanic_df['PassengerId'].count()

1309

How many people were on the titanic? 1309

How many woman and men survived?

In [9]:
survivors_df = titanic_df.copy()
survivors_df.groupby(['Sex','Survived'])['Sex'].count()

Sex     Survived
female  0            81
        1           385
male    0           734
        1           109
Name: Sex, dtype: int64

Survived 385 woman and 109 men 

What was the top 10 age group that survived the most and the top 10 age group that did not?

Top survivors

In [10]:
top_survivors_df = titanic_df.copy()
top_survivors_df = titanic_df.loc[titanic_df.Survived == 1]
top_survivors_df.groupby(['Age'])['Survived'].count().sort_values(ascending=False).head(10)

Age
22.0    21
24.0    20
36.0    16
30.0    16
18.0    16
27.0    15
35.0    14
29.0    13
19.0    12
45.0    12
Name: Survived, dtype: int64

top 10 not survivors

In [11]:
top_survivors_df = titanic_df.loc[titanic_df.Survived == 0]
top_survivors_df.groupby(['Age'])['Survived'].count().sort_values(ascending=False).head(10)

Age
21.0    33
25.0    27
24.0    27
30.0    24
28.0    23
18.0    23
22.0    22
26.0    20
29.0    17
19.0    17
Name: Survived, dtype: int64

How many positions or titles were on the boat?

In [12]:
class_df =titanic_df.copy()
class_df['Title'] = class_df.Name.apply(lambda name: name.split(',')[1].split('.')[0].strip())
class_df.groupby(['Title'])['PassengerId'].nunique().shape

(18,)

18

How much is the sum of the value of the tickets in USD

In [13]:
titanic_df['Fare'].sum()

43550.4869

43550.4869 USD